homeassistant/components/proximity/__init__.py

```diff
@@ -62,52 +42,22 @@
 )
 
 
-@callback
-def async_setup_proximity_component(
-    hass: HomeAssistant, name: str, config: ConfigType
-) -> bool:
-    """Set up the individual proximity component."""
-    ignored_zones: list[str] = config[CONF_IGNORED_ZONES]
-    proximity_devices: list[str] = config[CONF_DEVICES]
-    tolerance: int = config[CONF_TOLERANCE]
-    proximity_zone = config[CONF_ZONE]
-    unit_of_measurement: str = config.get(
-        CONF_UNIT_OF_MEASUREMENT, hass.config.units.length_unit
-    )
-    zone_friendly_name = name
-
-    proximity = Proximity(
-        hass,
-        zone_friendly_name,
-        DEFAULT_DIST_TO_ZONE,
-        DEFAULT_DIR_OF_TRAVEL,
-        DEFAULT_NEAREST,
-        ignored_zones,
-        proximity_devices,
-        tolerance,
-        proximity_zone,
-        unit_of_measurement,
-    )
-    proximity.entity_id = f"{DOMAIN}.{zone_friendly_name}"
-
-    proximity.async_write_ha_state()
-
-    async_track_state_change(
-        hass, proximity_devices, proximity.async_check_proximity_state_change
-    )
-
-    return True
-
-
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Get the zones and offsets from configuration.yaml."""
+    hass.data.setdefault(DOMAIN, {})
     for zone, proximity_config in config[DOMAIN].items():
-        async_setup_proximity_component(hass, zone, proximity_config)
+        _LOGGER.debug("setup %s with config:%s", zone, proximity_config)
+        coordinator = ProximityDataUpdateCoordinator(hass, zone, proximity_config)
+        await coordinator.async_config_entry_first_refresh()
```

Comment:
This isn't a config entry setup so we can't use this coordinator method.

Reply:
3ac7fc09475e8f2d31db690c3472809dd00a3646